src/SourceLocator/Type/PhpInternalSourceLocator.php

```diff
@@ -78,50 +74,45 @@ private function getInternalReflectionClass(Identifier $identifier) : ?Reflectio
             return null; // not an available internal class
         }
 
-        $reflection = new ReflectionClass($name);
+        $reflection = new CoreReflectionClass($name);
 
         return $reflection->isInternal() ? $reflection : null;
     }
 
-    /**
-     * Get the stub source code for an internal class.
-     *
-     * Returns null if nothing is found.
-     *
-     * @param string $className Should only contain [A-Za-z]
-     */
-    private function getStub(string $className) : ?string
+    private function getFunctionSource(Identifier $identifier) : ?InternalLocatedSource
     {
-        if (! $this->hasStub($className)) {
+        $functionReflection = $this->getInternalReflectionFunction($identifier);
+
+        if ($functionReflection === null) {
             return null;
         }
 
-        return file_get_contents($this->buildStubName($className));
-    }
+        $stub = $this->stubber->generateFunctionStub($functionReflection);
 
-    /**
-     * Determine the stub name
-     */
-    private function buildStubName(string $className) : ?string
-    {
-        if (! preg_match('/^[a-zA-Z_][a-zA-Z_\d]*$/', $className)) {
+        if ($stub === null) {
             return null;
         }
 
-        return __DIR__ . '/../../../stub/' . $className . '.stub';
+        return new InternalLocatedSource(
+            $stub,
+            $functionReflection->getExtensionName()
+        );
     }
 
-    /**
-     * Determine if a stub exists for specified class name
-     */
-    public function hasStub(string $className) : bool
```

Comment:
For future readers: this is indeed a BC break, but since this is an internal component (which is erroneously not marked as `@internal`), we can ignore it.